lib/happiness_delivered.rb

```diff
@@ -0,0 +1,111 @@
+# require 'customer_sms'
+require_relative 'sandcream'
+
+class HappinessDelivered
+  # include OrderConfirmation
+
+  attr_accessor :basket, :order_total, :menu
+
+  def initialize(basket = [])
+    @menu = Sandcream.new
+    @basket = basket
+    @order_total = []
+  end
+
+  def interactive_menu
+    loop do
+      print_menu_options
+      interface(STDIN.gets.chomp)
+    end
+  end
+  
+  def interface(customers_order)
+    case customers_order
+    when '1'
+      print_menu
+    when '2'
+      select_order
+    when '3'
+      review_order
+    when '4'
+      complete_order
+    when '5'
+      exit
+    else 
+      puts 'Please enter a valid menu option'
+    end
+  end
+
+  def print_menu
+    welcome_note
+    @menu.print_sando
+    @menu.print_icecream
+  end
+
+  def select_order
+    order
+    lines
+    puts 'Your current order:'
+    @basket.map { |dish| puts "• #{dish[:sando]}#{dish[:icecream]}" }
+  end
+
+  def review_order
+    lines
+    puts 'Your current order total:'
+    @basket.map { |price| puts "£ #{'%.2f' % price[:price]}" }
+    puts '-------'
+    @basket.map { |price| (@order_total << price[:price]) } 
+    puts "£ #{'%.2f' % @order_total.sum}"
+  end
+
+  def complete_order
+    # OrderConfirmation.send_sms
+    # bundle exec ruby customer_sms.rb
+    'magic message'
+  end
+
+  private
+
+  def lines
+    puts '-' * 60
+  end
+
+  def print_menu_options
+    lines
+    puts '1. View SandCreams menu'
+    puts '2. Select your order'
+    puts '3. Review your order'
+    puts '4. Complete your order'
+    puts '5. Exit menu'
+  end
+
+  def welcome_note
+    lines
+    puts 'Welcome to Happiness Delivered! Please select from' 
+    puts 'the following menu options to place your order.'
+    lines
+  end
+
+  def order
```

Comment:
Potentially, you could have a separate method for laying out the interface instructions for the user and call it here, so that in time you could refactor the instructions or reuse them elsewhere if other features were added